utils/python/CIME/system_test.py

```diff
@@ -98,40 +118,46 @@ def __init__(self, test_names,
             # Figure out what baseline name to use
-            if (baseline_name is None):
-                if(compare is not None and isinstance(compare,str)):
+            if baseline_name is None:
+                if compare is not None and isinstance(compare, str):
                     self._baseline_cmp_name = compare
                     self._compare = True
-                if(generate is not None and isinstance(generate,str)):
+                if generate is not None and isinstance(generate, str):
                     self._baseline_gen_name = generate
                     self._generate = True
                 branch_name = CIME.utils.get_current_branch(repo=self._cime_root)
-                expect(branch_name is not None, "Could not determine baseline name from branch, please use -b option")
-                if(self._compare and self._baseline_cmp_name is None):
+                expect(branch_name is not None,
+                       "Could not determine baseline name from branch, please use -b option")
+                if self._compare and self._baseline_cmp_name is None:
                     self._baseline_cmp_name = os.path.join(self._compiler, branch_name)
-                if(self._generate and self._baseline_gen_name is None):
+                if self._generate and self._baseline_gen_name is None:
                     self._baseline_gen_name = os.path.join(self._compiler, branch_name)
             else:
-                if(compare):
+                if compare:
                     self._compare = True
-                    self._baseline_cmp_name  = baseline_name
-                    if (not self._baseline_cmp_name.startswith("%s/" % self._compiler)):
-                        self._baseline_cmp_name = os.path.join(self._compiler, self._baseline_cmp_name)
-                if(generate):
+                    self._baseline_cmp_name = baseline_name
+                    if not self._baseline_cmp_name.startswith("%s/" % self._compiler):
+                        self._baseline_cmp_name = os.path.join(self._compiler,
+                                                               self._baseline_cmp_name)
+                if generate:
                     self._generate = True
                     self._baseline_gen_name  = baseline_name
-                    if (not self._baseline_gen_name.startswith("%s/" % self._compiler)):
-                        self._baseline_gen_name = os.path.join(self._compiler, self._baseline_gen_name)
+                    if not self._baseline_gen_name.startswith("%s/" % self._compiler):
+                        self._baseline_gen_name = os.path.join(self._compiler,
+                                                               self._baseline_gen_name)
 
             # Compute baseline_root
-            self._baseline_root = baseline_root if baseline_root is not None else self._machobj.get_value("CCSM_BASELINE")
-            if (self._project is not None):
+            if baseline_root is None:
```

Comment:
I always run pylint with "--disable C" (disables all convention checks) for this reason. In my opinion, the conciseness of the ternary is too useful to throw out due to an arbitrary line length limit.